File: src/main/java/org/intellij/sdk/language/jass/JassFileType.java
package org.intellij.sdk.language.jass;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class JassFileType extends LanguageFileType {

    public static final JassFileType INSTANCE = new JassFileType();

    private JassFileType() {
        super(JassLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "JASS File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "JASS language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "j";
    }

    @Override
    public @NotNull Icon getIcon() {
        return JassIcons.FILE;
    }
}

File: src/main/kotlin/raft/war/language/jass/openapi/fileTypes/JassFileType.kt
package raft.war.language.jass.openapi.fileTypes

import com.intellij.openapi.fileTypes.LanguageFileType
import raft.war.language.jass.icons.JassIcons
import raft.war.language.jass.lang.JassLanguage
import javax.swing.Icon

// https://plugins.jetbrains.com/docs/intellij/language-and-filetype.html

class JassFileType : LanguageFileType(JassLanguage.instance) {

    override fun getName(): String = JassLanguage.NAME
    override fun getDescription(): String = JassLanguage.NAME
    override fun getDefaultExtension(): String = EXTENSION
    override fun getIcon(): Icon = JassIcons.FILE

    @Suppress("CompanionObjectInExtension")
    companion object {
        val instance: JassFileType = JassFileType()

        const val EXTENSION: String = "j"
    }
}
